src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php

```diff
@@ -217,18 +268,54 @@ public function getValidationRuleForGetAll(): ParamRuleCollection
      *                 property="data",
      *                 ref="#/components/schemas/Claim-RequestModel"
      *             ),
-     *             @OA\Property(property="meta", type="object")
+     *             @OA\Property(
+     *                 property="meta",
+     *                 type="object",
+     *                 @OA\Property(property="allowedActions", type="array", @OA\Items(type="object"))
+     *             )
      *         )
      *     )
      * )
      * @inheritDoc
      */
     public function getOne(): EndpointResult
     {
-        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
-        $claimRequest = $this->getClaimService()->getClaimDao()->getClaimRequestById($id);
+        $id = $this->getRequestParams()
+            ->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $empNumber = $this->getEmpNumber();
+
+        if (!$this->checkLoggedInUser($empNumber)) {
+            throw $this->getForbiddenException();
+        }
+
+        if (!$this->getUserRoleManagerHelper()->isEmployeeAccessible($empNumber)) {
+            throw $this->getForbiddenException();
+        }
```

Comment:
*

Reply:
is supervisor accessible to a ESS - this type of scenarios cannot handle without this